operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -842,12 +837,15 @@ func (r *StackReconciler) markStackSucceeded(ctx context.Context, instance *pulu
 		Permalink:            shared.Permalink(update.Status.Permalink),
 		LastResyncTime:       metav1.Now(),
 	}
+	if last != nil && last.Generation == current.Generation {
+		instance.Status.LastUpdate.Failures = last.Failures
+	}
```

Comment:
I would advocate for resettting the failure count to zero here. Once the update is successful, retaining the failure count seems debatable.  A couple of weird cases:
1. In the case of period resync, the failure count stays at non-zero in perpetuity.
2. In the case of sporatic errors (fail, success, fail), one might use backoff prematurely.

When a container recovers from a crashloop, does the status still reflect the situation or are "events" the only historical record at that point?

Reply:
> In the case of sporatic errors (fail, success, fail), one might use backoff prematurely. Or is that intentional?

Very good point, this isn't intention. We should definitely reset here.